packages/h5wasm/src/utils.ts

```diff
@@ -97,3 +125,15 @@ export function convertSelectionToRanges(
     return [Number(member), Number(member) + 1];
   });
 }
+
+export function convertToBool(value: unknown): boolean | BoolArray {
+  if (Array.isArray(value)) {
+    return value.map(convertToBool);
+  }
+
+  if (isTypedArrayLodash(value)) {
+    return [...(value as TypedArray)].map(convertToBool);
```

Comment:
Actually, this poses an issue for attribute display as the `Int8Array` returned are always 1D. even if the underlying dataset has more dimensions. This means nD boolean arrays as attributes will be displayed as 1D arrays.

I didn't find an easy way to fix this.

Reply:
Is there an expectation that a library will return something with more dimensions?  Given that JS/TS only has 1d typed arrays, how would this look?

Reply:
It could return regular (not typed) arrays so that multiple dimensions are accounted by nested arrays

Reply:
There are performance (and other) issues with using nested arrays instead of typed arrays.  I still don't understand the scope of the request - is it preferred in your use case that all dataset and attribute values be converted to nested arrays, or just attribute values, or just "boolean" attribute values?

Reply:
Let's continue the discussion on https://github.com/usnistgov/h5wasm/issues/25